authors/apps/friends/views.py

```diff
@@ -0,0 +1,135 @@
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from .models import Friend
+from .serializers import FollowersFollowingSerializer, FollowUnfollowSerializer
+
+
+class FollowUnfollowApiView(generics.RetrieveUpdateDestroyAPIView):
+    """view for following a user"""
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, username, format=None):
+        """
+        Checks if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, returned error message
+        Else create the user relationship
+        Finally return the serialized object of the followed user
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot follow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        if Friend.objects.filter(user_from=follower, user_to=followed):
```

Comment:
See my comment on `get or create` below

Reply:
This line is not required. get_or_create returns a tuple that tells you if that relationship exists previously. If it does, return this response otherwise continue with the rest of the code.